pkg/clients/gitlab/client.go

```diff
@@ -0,0 +1,51 @@
+package gitlab
+
+import (
+	"fmt"
+
+	"github.com/redhat-data-and-ai/usernaut/pkg/common/structs"
+	"github.com/redhat-data-and-ai/usernaut/pkg/utils"
+	gitlab "gitlab.com/gitlab-org/api/client-go"
+)
+
+type GitlabClient struct {
+	gitlabClient  *gitlab.Client
+	ldapSync      bool
+	externalTeams map[string]*structs.Team // Teams from external backends (like Rover)
+}
+
+type GitlabConfig struct {
+	URL      string `json:"url"`
+	Token    string `json:"token"`
+	LdapSync bool   `json:"ldap_sync"`
```

Comment:
should we also take a projectID as input under which we want to create the groups? This field can be optional. If passed, groups are created there but if not passed then groups are created globally.